cluster/kubernetes/manifests.go

```diff
@@ -8,18 +8,31 @@ import (
 )
 
 type Manifests struct {
+	AllowedNamespaces []string
 }
 
 func (c *Manifests) LoadManifests(base string, paths []string) (map[string]resource.Resource, error) {
-	return kresource.Load(base, paths)
+	resources, err := kresource.Load(base, paths)
+	c.discardDisallowed(resources)
+	return resources, err
 }
 
 func (c *Manifests) ParseManifests(allDefs []byte) (map[string]resource.Resource, error) {
-	return kresource.ParseMultidoc(allDefs, "exported")
+	resources, err := kresource.ParseMultidoc(allDefs, "exported")
+	c.discardDisallowed(resources)
+	return resources, err
 }
 
 func (c *Manifests) UpdateImage(def []byte, id flux.ResourceID, container string, image image.Ref) ([]byte, error) {
 	return updatePodController(def, id, container, image)
 }
 
+func (c *Manifests) discardDisallowed(resources map[string]resource.Resource) {
+	for k, r := range resources {
+		if !isInAllowedNamespace(c.AllowedNamespaces, r.ResourceID()) {
```

Comment:
It's a bit ugly doing this when parsing (parsing should be parsing, not filtering) but the only alternative I saw was passing `AllowedNamespaces` all over the place which was uglier and more error-prone (people using new code may forget filtering by namespace where needed).

Reply:
Yeah I have to say I don't like this either -- these are supposed to be general-purpose. Can I give an example where filtering the return value will screw things up? Not off the top of my head, but I think it's about as likely that will be a problem as the scenario you mention of someone forgetting to filter before an operation.

Reply:
Then, the alternative is to filter after loading the manifests (probably systematically after every call). Would you be OK with that?

> Can I give an example where filtering the return value will screw things up? Not off the top of my head

Me neither (it's like the manifests didn't exist on the first place), that's why I chose this approach.